pkg/extprom/http/instrument_server.go

```diff
@@ -0,0 +1,107 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// ServerInstrumentor holds necessary metrics to instrument an http.Server
+// and provides necessary behaviors
+type ServerInstrumentor interface {
+	// NewInstrumentedHandler wraps the given HTTP handler for instrumentation.
+	NewInstrumentedHandler(handlerName string, handler http.Handler) http.HandlerFunc
+	// NewInstrumentedHandlerFunc wraps the given function for instrumentation.
+	NewInstrumentedHandlerFunc(handlerName string, next func(http.ResponseWriter, *http.Request)) http.HandlerFunc
```

Comment:
I would vote for simpler interface and removal of one of those methods. This one is to shallow IMO and there are many easy ways to tranform `handler` to `next func` and vice versa. (: 

Let's keep one way of doing things.